### PR TITLE
Save_variables fun key exclusion

### PR DESCRIPTION
**Pull request by ew3361zh**
This PR seeks to address issue #524 to expand the save_variable function in OM to handle exclusionary filter lists of requested variables in addition to the previous inclusionary filter lists.

# To accomplish this, the PR:
1. Updates `_filter_variables_pool` to filter by either exclusion and inclusion by checking the value of the first filter key in a given filter document and acting appropriately.
2. Adds an example exclusion filter file to the output_filters directory.
3. Updates unit testing across the board.

### CODE REVIEW

**Review by PooyaHekmati — CHANGES_REQUESTED** [review]
Hi Niko, I requested some changes in Slack but didn't update them here. I'm requesting changes here so you can re-request review when you are done adding those.

File: RUFAS/output_manager.py [hunk 1]
```diff
@@ -373,7 +373,12 @@ def _filter_variables_pool(self, inclusion_keys: List[str]) -> Dict[str, pool_el
             from the inclusion_keys list remaining from the variables_pool.
 
         """
-        return {key: self.variables_pool[key] for key in inclusion_keys if key in self.variables_pool.keys()}
+        exclude_keyword_location = 0
+        exclude_keyword = "exclude"
```
Comment (PooyaHekmati) [inline RUFAS/output_manager.py:383]: nice!

File: RUFAS/output_manager.py [hunk 2]
```diff
@@ -356,14 +356,14 @@ def _load_txt_file_to_list(self, path: str) -> List[str]:
         except Exception as e:
             raise e
 
-    def _filter_variables_pool(self, inclusion_keys: List[str]) -> Dict[str, pool_element_type]:
+    def _filter_variables_pool(self, filter_keys: List[str]) -> Dict[str, pool_element_type]:
         """
-        Takes the list of keys the user wants in their final data pool,
+        Takes the list of keys the user wants to their final data pool either by inclusion or exclusion,
```
Comment (PooyaHekmati) [inline RUFAS/output_manager.py]: This sentence doesn't sound right. 
Can we rephrase it to something like: returns a filtered variables pool based on either inclusion or exclusion?

Also, the fact that the first element of the list is important is not mentioned in the docstring.

**Review by PooyaHekmati — APPROVED** [review]
[LGTM](https://lgtm.lol/i/758)

**Review by JoeWaddell — APPROVED** [review]
This looks great! Some clever work in here, and I've tested it a number of ways and it works nicely. Well done
